### PR TITLE
Make ObjectMappers methods return copy of mappers

### LINKED ISSUE #151: ObjectMappers methods return shared mutable ObjectMapper
`ObjectMappers.vanilla()` and `ObjectMappers.guavaJdk7()` both directly return a static variable that is a mutable object, which means that if a caller changes the configuration of the returned object, the change is shared across all instances and for all future callers.

These methods should be changed to hand back a copy of the mappers instead so that callers are free to mutate the returned values without impacting other instances or future callers.

### PR DESCRIPTION
**Pull request by nmiyake**
Fixes #151

### CODE REVIEW

**Comment by nmiyake** [comment]
@uschi2000 @markelliot

**Comment by markelliot** [comment]
urgh, I think we need to find a way to make these immutable rather than pass around copies; we were trying to control memory load a bit by sharing objects.

**Comment by nmiyake** [comment]
Hmm, in that case, maybe it would make sense to make the shared instances package-protected and keep public methods that hand back copies? Just seems pretty dangerous/error-prone to have a key object open to modification from anywhere within a JVM.

`ObjectMapper` is non-final so it's technically possible to make an unmodifiable wrapper, but that would be a bigger change with more possible implications (especially if someone does actually want a copy that they can mutate/configure for their own use)

**Comment by uschi2000** [comment]
I like this idea: "sense to make the shared instances package-protected and keep public methods that hand back copies"

**Comment by nmiyake** [comment]
Implemented suggested variation. Made it so that static variables are package-protected and the internal methods use the variables directly -- I think this makes it much more apparent when reading code that these are shared instances.

**Comment by uschi2000** [comment]
Will look tomorrow

File: http-clients/src/main/java/com/palantir/remoting/http/ObjectMappers.java [hunk 1]
```diff
@@ -22,32 +22,28 @@
...
 public final class ObjectMappers {
 
-    private static final ObjectMapper VANILLA_MAPPER = new ObjectMapper();
+    static final ObjectMapper VANILLA_MAPPER = vanilla();
+    static final ObjectMapper GUAVA_JDK7_MAPPER = guavaJdk7();
 
-    private static final ObjectMapper GUAVA_JDK7_MAPPER;
+    private ObjectMappers() {}
+
+    public static ObjectMapper vanilla() {
```
Comment (markelliot) [inline http-clients/src/main/java/com/palantir/remoting/http/ObjectMappers.java:42]: - some Java doc?

File: http-clients/src/main/java/com/palantir/remoting/http/ObjectMappers.java [hunk 2]
```diff
@@ -22,32 +22,28 @@
 
 public final class ObjectMappers {
 
-    private static final ObjectMapper VANILLA_MAPPER = new ObjectMapper();
+    static final ObjectMapper VANILLA_MAPPER = vanilla();
+    static final ObjectMapper GUAVA_JDK7_MAPPER = guavaJdk7();
```
Comment (markelliot) [inline http-clients/src/main/java/com/palantir/remoting/http/ObjectMappers.java:35]: Worth adding some comments here

**Comment by nmiyake** [comment]
Added comments

**Pull request merged**